src/core/lib/draw-layers.js

```diff
@@ -231,24 +229,20 @@ function getViewportFromDescriptor(viewportOrDescriptor) {
  * Returns the picking color of currenlty selected object of the given 'layer'.
  * @return {Array} - the picking color or null if layers selected object is invalid.
  */
-function updateLayerHighlightColor(layer) {
+function getLayerHighlightColor(layer) {
```

Comment:
With this change, the name is no longer as instructive: `getPickingModuleParameters`?

Reply:
In this particular case, we are indeed only populating module settings when needed.

Generally speaking, I prefer sending all overridable moduleSettings/uniforms/parameters to `draw`. Currently we have multiple `withParameters` wrapped around the draw call and it's very difficult to track the context when debugging.